provider/pkg/gen/schema.go

```diff
@@ -145,6 +145,11 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 
 	definitions := swagger["definitions"].(map[string]interface{})
 	groupsSlice := createGroups(definitions)
+
+	// Make a copy of the overlay types and resources
+	overlayTypes := pkg.Types
```

Comment:
The merge logic looks fine to me, but I was initially confused that the overlays were being defined on `pkg.Types` and `pkg.Resources`. I think it might be clearer to pull overlays out into a separate map, perhaps defined in an overlays file to make it clearer that we're augmenting the schema generated off the k8s spec.